### PR TITLE
Add cmdline driver test

### PR DESCRIPTION
**Pull request by adrianchiris**

### CODE REVIEW

**Comment by adrianchiris** [comment]
need to review PR 21 first, i will then rebase this one

File: pkg/tc/driver/cmdline/tc_cmdline_test.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
+package driver_test
+
+import (
+	"github.com/Mellanox/multi-networkpolicy-tc/pkg/policyrules"
```
Comment (ykulazhenkov) [inline pkg/tc/driver/cmdline/tc_cmdline_test.go:4]: import order

File: pkg/tc/driver/cmdline/tc_cmdline_test.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+				WithMatchKeyIPProto(string(policyrules.ProtocolTCP)).
+				WithAction(tctypes.NewGenericActionBuiler().WithPass().Build()).
+				Build()
+
+			filters, err := tcCmdLine.FilterList(ingressQdisc)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fakeCmd.Argv).To(BeEquivalentTo(expectedCmdArgs))
+			Expect(filters).To(HaveLen(1))
+			Expect(filters[0].Equals(expectedFilter)).To(BeTrue())
```
Comment (ykulazhenkov) [inline pkg/tc/driver/cmdline/tc_cmdline_test.go:402]: is To(BeTrue()) part required?

File: pkg/tc/driver/cmdline/tc_cmdline_test.go [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+		})
+
+		It("retuns error when underlying command errors", func() {
+			fakeCmd.OutputScript = append(fakeCmd.OutputScript, newFakeAction(
+				nil, nil, errors.New("test error!")))
```
Comment (ykulazhenkov) [inline pkg/tc/driver/cmdline/tc_cmdline_test.go:407]: nit: you are using `errors.New("test error!")` in many tests. Maybe move it to a errTest var or something?

File: pkg/tc/driver/cmdline/tc_cmdline_test.go (hunk 1 above)
Comment (adrianchiris) [reply pkg/tc/driver/cmdline/tc_cmdline_test.go:4]: i really need to check how to sort it in ide

File: pkg/tc/driver/cmdline/tc_cmdline_test.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/tc/driver/cmdline/tc_cmdline_test.go:402]: in this test i want to validate that FilterList outputs expected filters, so id like to compare it to an actual filter that is expected to be returned.

Expect call returns an Assertion which you need to check with a matcher so To(BeTrue()) is required

Comment (ykulazhenkov) [reply pkg/tc/driver/cmdline/tc_cmdline_test.go:402]: Sorry, I misunderstood this construction. For some reason I thought Equals was part of the assertion check.
False positive, sorry for wasting your time

File: pkg/tc/driver/cmdline/tc_cmdline_test.go (hunk 3 above)
Comment (adrianchiris) [reply pkg/tc/driver/cmdline/tc_cmdline_test.go:407]: Done.

**Review by ykulazhenkov — APPROVED** [review]

**Pull request merged**